src/__tests__/persistCache.ts

```diff
@@ -0,0 +1,155 @@
+      } catch (e) {
+        expect(() => {
+          throw e;
+        }).toThrowErrorMatchingSnapshot();
+      }
+    });
+    it('requires storage', async () => {
+      try {
+        await persistCache({ cache: new InMemoryCache() });
+        fail('invoking persistCache without storage should throw an error');
+      } catch (e) {
+        expect(() => {
+          throw e;
+        }).toThrowErrorMatchingSnapshot();
+      }
+    });
+  });
+
+  describe('basic usage', () => {
+    const operation = gql`
+      {
+        hello
+      }
+    `;
+    const result = { data: { hello: 'world' } };
+
+    it('extracts a previously filled InMemoryCache from storage', async () => {
+      const [client, client2] = await simulateApp({
+        operation,
+        result,
+      });
+      expect(client.extract()).toEqual(client2.extract());
+    });
+    it('extracts a previously filled HermesCache from storage', async () => {
+      const [client, client2] = await simulateApp({
+        operation,
+        result,
+        persistOptions: {
+          cache: new Hermes(),
+        },
+      });
+      expect(client.extract()).toEqual(client2.extract());
+    });
+  });
+
+  describe('nested queries', () => {
+    const operation = gql`
+      {
+        user(id: 1) {
+          name {
+            last
+            first
+          }
+          posts {
+            title
+            comments {
+              name
+            }
+          }
+        }
+      }
+    `;
+    const result = {
+      data: {
+        user: {
+          name: { last: 'Doe', first: 'Jane', __typename: 'Name' },
+          posts: [
+            {
+              title: 'Apollo is awesome',
+              comments: [{ name: 'foo', __typename: 'Comment' }],
+              __typename: 'Post',
+            },
+          ],
+          __typename: 'User',
+        },
+      },
+    };
+
+    it('extracts a previously filled InMemoryCache from storage', async () => {
+      const [client, client2] = await simulateApp({
+        operation,
+        result,
+      });
+      expect(client.extract()).toEqual(client2.extract());
+    });
+    it('extracts a previously filled HermesCache from storage', async () => {
+      const [client, client2] = await simulateApp({
+        operation,
+        result,
+        persistOptions: {
+          cache: new Hermes(),
+        },
+      });
+      expect(client.extract()).toEqual(client2.extract());
+    });
+  });
+
+  describe('advanced usage', () => {
+    const operation = gql`
+      {
+        hello
+      }
+    `;
+    const result = { data: { hello: 'world' } };
+
+    it('passing in debounce configures the debounce interval', async () => {
+      const debounce = 600;
+      const storage = new MockStorage();
+
+      await simulateWrite({ debounce, storage, result, operation });
+
+      expect(await storage.getItem('apollo-cache-persist')).toBe(undefined);
```

Comment:
`apollo-cache-persist` is a bit of a magic value here (since we don't expose it).

I feel like it'd be less fragile to use Jest mocks for this (i.e., check to see if the `setItem` method has been invoked), but I'm not super familiar with Jest mocking. So... unless that approach immediately makes sense to you, this is fine :)

Reply:
I'll give it a shot! I'm not a Jest mocking expert either so hopefully this isn't too hard haha.